modules/forum_pinner.js

```diff
@@ -0,0 +1,78 @@
+const { SlashCommandBuilder } = require("@discordjs/builders");
+
+async function pinPost(thread) {
+  let messages = await thread.messages.fetch();
+  let firstMessage = messages.first();
+
+  if(firstMessage.pinned)
+    return;
+
+  return await firstMessage.pin();
+}
+
+module.exports = {
+  enabled: true,
+
+  name: "Forum Pinner",
+  description: "Automatically pins the first message of a forum post for easy retrieval.",
+
+  commands: [
+    {
+      data: new SlashCommandBuilder()
+        .setName("pin")
+        .setDescription("Pin the first message of the post!"),
+
+      async execute(interaction) {
+
+        if(!interaction.memberPermissions.has("MANAGE_MESSAGES")) {
```

Comment:
I'm not entirely familiar with forum permissions; does this have any special interaction for the thread owner, or is it expected to only be a staff command?

Reply:
That function checks whether the member that executed the `/pin` command has the `MANAGE_MESSAGES` permission.
It is a staff command.